gammapy/time/lightcurve.py

```diff
@@ -314,11 +321,204 @@ def create_fixed_time_bin(time_step, spectrum_extraction):
+            emin = e_reco[e_reco.searchsorted(max(spec.lo_threshold, energy_range[0]))]
+            emax = e_reco[e_reco.searchsorted(min(spec.hi_threshold, energy_range[1])) - 1]
+            # filter the event list with energy
+            on = on.select_energy([emin, emax])
+            on = on.select_energy(energy_range)
+            off = off.select_energy([emin, emax])
+            off = off.select_energy(energy_range)
+        if interval is not None:
+            # filter the event list with time
+            tmin = interval[0]
+            tmax = interval[1]
+            on = on.select_time([tmin, tmax])
+            off = off.select_time([tmin, tmax])
+        if extra:
+            return on, off, spec, e_reco
+        return on, off
+
+    @staticmethod
+    def _alpha(time_holder, obs_properties, n, istart, i):
+        """ Helper function for make_time_intervals_min_significance
+
+        Parameters
+        ----------
+        time_holder : `list` of float and flag
+            Contains a list of a time and a flag in 2-element arrays
+        obs_properties : `list` of dictionaries
+            Contains the dead time fraction and ratio of the on/off region
+        n : int
+            First observation to use
+        istart : int
+            index of the start of the interval
+        i : int
+            index of the end of the interval
+        """
+
+        def in_list(item, L):
+            o, j = np.where(L == item)
+            for index in o:
+                yield index
+
+        alpha = 0
+        tmp = 0
+        time = 0
+        xm1 = istart
+        # loop over observations
+        for x in in_list('end', time_holder[istart:i + 1]):
+            alpha += (1 - obs_properties['deadtime'][n + tmp]) * (
+                    float(time_holder[x][0]) - float(time_holder[xm1][0])) * \
+                     obs_properties['A_off'][n + tmp]
+            time += (1 - obs_properties['deadtime'][n + tmp]) * (float(time_holder[x][0]) - float(time_holder[xm1][0]))
+            xm1 = x + 1
+            tmp += 1
+        alpha += (1 - obs_properties['deadtime'][n + tmp]) * (float(time_holder[i][0]) - float(time_holder[xm1][0])) * \
+                 obs_properties['A_off'][n + tmp]
+        time += (1 - obs_properties['deadtime'][n + tmp]) * (float(time_holder[i][0]) - float(time_holder[xm1][0]))
+        alpha = time / alpha
+        return alpha
+
+    def make_time_intervals_min_significance(self, significance, significance_method, energy_range,
+                                             spectrum_extraction, separators=None):
+        """
+
+        Create time intervals such that each bin of a light curve reach a given significance
+
+        The function create a list of time associated with identifiers called time_holder.
+        The identifiers can be 'on' for the on events, 'off' for the off events, 'start' for the start of an
+        observation, 'end for the end of an observation and 'break' for a separator.
```

Comment:
The docstring is what users see. So ideally, here you would give a high-level description of the algorithm, but not describe implementation details e.g. that there's an internal variable `time_holder`. For such a complex algorithm it can be useful to have a few line of comments with `# ...` at the start of the function that describe the idea, i.e. that the central data structure you create and operate on is the `time_holder` list.

Some small suggestions:

- Make `separators` a `Time` object? And better describe what it's purpose is in the docstring for users. How would one usually use it? Does it force breaks even if the desired significance is not reached?

- Rename `time_holder` variable to `events`?

Reply:
I can make it a TIme object. Changed the docstring. It seem there is a confusion, a bin can't reach the wanted significance and a break, so the desired significance is never reached. The current point is discarded when reaching a 'break' element.

Not all element of time_holder is an event so I'm not for this change.